### PR TITLE
Used AlertDialog instead of MaterialDialog in NoteEditor.kt

### PR DESCRIPTION
**Pull request by SanjaySargam**
## Fixes
Related #13315 

## How has been Tested?

Tested on Emulator
WhatsApp Image 2023-03-22 at 3 54 18
WhatsApp Image 2023-03-22 at 3 54 18 PM

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
Do you have a screenshot?

and as a note, please don't delete the `How Has This Been Tested?` section and try filling it.

**Comment by SanjaySargam** [comment]
> and please don't delete the `How Has This Been Tested?` section and try filling it.

Ok Will take care next time

**Comment by SanjaySargam** [comment]
> Do you have a screenshot?


@BrayanDSO 

WhatsApp Image 2023-03-22 at 3 54 18
WhatsApp Image 2023-03-22 at 3 54 18 PM

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 1]
```diff
@@ -30,7 +30,6 @@ import android.net.Uri
 import android.os.Build
 import android.os.Bundle
 import android.text.Editable
-import android.text.TextUtils.isEmpty
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:33]: interesting - this is unrelated, but at the same time it appears to be unused if this is still compiling correctly, so it is a valid change.

Should be in a separate lint-labeled commit for 💯 % credit tho ;-)

**Review by mikehardy — APPROVED** [review]
LGTM

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 2]
```diff
@@ -1802,12 +1798,12 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
 
     private fun suggestRemoveButton(
         button: CustomToolbarButton,
-        editToolbarItemDialog: MaterialDialog
+        editToolbarItemDialog: AlertDialog.Builder
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:1801]: The type of `editToolbarItemDialog` should be `AlertDialog` so below you can call `dismiss()` on it. The change between `dismiss()` and `cancelable(true)` doesn't make any sense and it doesn't reproduce the original code's  intent.

**Review by lukstbit — CHANGES_REQUESTED** [review]
The changes introduces in this PR completely break the functionality of the changed dialogs. I would recommend to start with the change I mention below and from there make the proper changes so the intent of the original code is maintained.

Note that there should also be some padding added in the layout used as a custom view because the original UI isn't reproduced without it.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt (hunk 2 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:1801]: If I initialised `editToolbarItemDialog` as a `AlertDialog` then its affecting the other dialog and give error

**Review by mikehardy — CHANGES_REQUESTED** [review]
Does not match behavior of previous code

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt (hunk 2 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:1801]: > If I initialised `editToolbarItemDialog` as a `AlertDialog` then its affecting the other dialog and giving error

Yes, there will be errors, as this isn't going to be a simple change method names/ imports and be done. The type there needs to be AlertDialog so you can dismiss it. It shouldn't be very difficult to do.

**Comment by SanjaySargam** [comment]
> There are still conflicts...

Can you plz elaborate  the conflicts which are there

**Comment by lukstbit** [comment]
There's a conflict in the imports of NoteEditor. If you look at the bottom of this discussion the Github UI tells you that there's a conflict.

**Comment by SanjaySargam** [comment]
> There's a conflict in the imports of NoteEditor. If you look at the bottom of this discussion the Github UI tells you that there's a conflict.

How these conflicts can be resolved

**Comment by lukstbit** [comment]
I fixed the conflict, you can see the commit for more information. To avoid these conflicts in the future you could follow these steps before pushing:
- update the main branch with the latest changes from this repository(this is particularly important when the PR is older then a few days(the longer a PR stays open the greater chances of a conflict due to code that was merged))
- rebase your PR on top of main, these would allow you to see and fix any possible conflicts
- as an extra step, if the amount of changes is small(like in this PR) you could squash them into a single commit
- push the code to the PR branch.

**Comment by SanjaySargam** [comment]
> I fixed the conflict, you can see the commit for more information. To avoid these conflicts in the future you could follow these steps before pushing:
> - update the main branch with the latest changes from this repository(this is particularly important when the PR is older then a few days(the longer a PR stays open the greater chances of a conflict due to code that was merged))
> - rebase your PR on top of main, these would allow you to see and fix any possible conflicts
> - as an extra step, if the amount of changes is small(like in this PR) you could squash them into a single commit
> - push the code to the PR branch.
> 

Ok Got it. Thanks ♥️

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 3]
```diff
@@ -1817,20 +1813,22 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
...
+        get() = AlertDialog.Builder(this)
             .neutralButton(R.string.help) {
                 openUrl(Uri.parse(getString(R.string.link_manual_note_format_toolbar)))
             }
             .negativeButton(R.string.dialog_cancel)
 
     private fun displayAddToolbarDialog() {
+        val v = layoutInflater.inflate(R.layout.note_editor_toolbar_add_custom_item, null)
+        v.setPadding(50, 0, 50, 0)
+        val view = ScrollView(this)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:1826]: This ScrollView needs to be added in the layout R.layout.note_editor_toolbar_add_custom_item wrapping the top LinearLayout. Add a standard padding to it of 16dp in all directions.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 4]
```diff
@@ -1841,6 +1839,9 @@ class NoteEditor : AnkiActivity(), DeckSelectionListener, SubtitleListener, Tags
 
     private fun displayEditToolbarDialog(currentButton: CustomToolbarButton) {
         val view = layoutInflater.inflate(R.layout.note_editor_toolbar_edit_custom_item, null)
+        view.setPadding(50, 0, 50, 0)
+        val v = ScrollView(this)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:1843]: As above this ScrollView needs to be added in the layout R.layout.note_editor_toolbar_edit_custom_item wrapping the top LinearLayout. Add a standard padding to it of 16dp in all directions.

**Review by lukstbit — CHANGES_REQUESTED** [review]
Almost there...

**Review by lukstbit — APPROVED** [review]
Great!

**Review by mikehardy — APPROVED** [review]
Now it is looking good, thank you! Thanks also for sure to @lukstbit - big kudos for helping work this PR through

**Pull request merged**